executor/runner/runner.go

```diff
@@ -181,9 +173,7 @@ func (r *Runner) runMainContainerAndStartSidecars(ctx context.Context, startTime
 	default:
 	}
 	r.maybeSetDefaultTags(ctx) // initialize metrics.Reporter default tags
-	pod, podLock := r.container.Pod()
-	containerNames := getContainerNames(pod)
-	podLock.Unlock()
+	containerNames := r.container.OrderSortedContainerNames()
 	startingMsg := fmt.Sprintf("starting %d container(s): %s", len(containerNames), containerNames)
```

Comment:
startingMsg := fmt.Sprintf("starting %d container(s) in order: %s", len(containerNames), containerNames)